src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex.cs

```diff
@@ -61,9 +61,16 @@ protected AbstractSyntaxIndex(Checksum? checksum)
                 return null;
 
             // Populate our caches with this data.
-            s_documentToIndex.GetValue(document, _ => index);
+#if NET
+            s_documentToIndex.TryAdd(document, index);
+            s_documentIdToIndex.AddOrUpdate(document.Id, index);
+#else
+            // Avoid capturing index on the fast path by making a copy for the slow path
```

Comment:
>// Avoid capturing index on the fast path by making a copy for the slow path [](http://example.com/codeflow?start=12&length=76)

Could this be moved to a local fn to avoid the allocation instead?

Reply:
Creating a temporary local is pretty standard practice for this.